spec/lib/saml_idp/signed_info_builder_spec.rb

```diff
@@ -26,37 +29,38 @@ module SamlIdp
       expect(subject.signed).to eq("hKLeWLRgatHcV6N5Fc8aKveqNp6Y/J4m2WSYp0awGFtsCTa/2nab32wI3du+3kuuIy59EDKeUhHVxEfyhoHUo6xTZuO2N7XcTpSonuZ/CB3WjozC2Q/9elss3z1rOC3154v5pW4puirLPRoG+Pwi8SmptxNRHczr6NvmfYmmGfo=")
     end
 
-    it "builds a legit raw XML file when using cloudhsm" do
-      mock_cloudhsm
-      expect(subject.raw).to eq("<ds:SignedInfo xmlns:ds=\"http://www.w3.org/2000/09/xmldsig#\"><ds:CanonicalizationMethod Algorithm=\"http://www.w3.org/2001/10/xml-exc-c14n#\"></ds:CanonicalizationMethod><ds:SignatureMethod Algorithm=\"http://www.w3.org/2001/04/xmldsig-more#rsa-sha256\"></ds:SignatureMethod><ds:Reference URI=\"#_abc\"><ds:Transforms><ds:Transform Algorithm=\"http://www.w3.org/2000/09/xmldsig#enveloped-signature\"></ds:Transform><ds:Transform Algorithm=\"http://www.w3.org/2001/10/xml-exc-c14n#\"></ds:Transform></ds:Transforms><ds:DigestMethod Algorithm=\"http://www.w3.org/2001/04/xmlenc#sha256\"></ds:DigestMethod><ds:DigestValue>em8csGAWynywpe8S4nN64o56/4DosXi2XWMY6RJ6YfA=</ds:DigestValue></ds:Reference></ds:SignedInfo>")
-    end
+    context 'with cloudhsm key label' do
+      before do
+        mock_cloudhsm
+      end
 
-    it "builds a legit digest of the XML file when using cloudhsm" do
-      mock_cloudhsm
-      expect(subject.signed).to eq('')
-    end
+      subject do
+        described_class.new(
+          reference_id,
+          digest,
+          algorithm,
+          cloudhsm_key_label: 'secret',
+          secret_key: Default::SECRET_KEY
+        )
+      end
 
-    it "raises key not found when the cloudhsm key label is not found" do
-      mock_cloudhsm
-      allow(MockSession).to receive_message_chain(:find_objects, :first).and_return(nil)
-      stub_const 'SamlIdp::Default::SECRET_KEY', 'secret'
-      expect { subject.signed }.to raise_error(RuntimeError, 'cloudhsm key not found for label: secret')
-    end
+      it "builds a legit raw XML file when using cloudhsm" do
+        expect(subject.raw).to eq("<ds:SignedInfo xmlns:ds=\"http://www.w3.org/2000/09/xmldsig#\"><ds:CanonicalizationMethod Algorithm=\"http://www.w3.org/2001/10/xml-exc-c14n#\"></ds:CanonicalizationMethod><ds:SignatureMethod Algorithm=\"http://www.w3.org/2001/04/xmldsig-more#rsa-sha256\"></ds:SignatureMethod><ds:Reference URI=\"#_abc\"><ds:Transforms><ds:Transform Algorithm=\"http://www.w3.org/2000/09/xmldsig#enveloped-signature\"></ds:Transform><ds:Transform Algorithm=\"http://www.w3.org/2001/10/xml-exc-c14n#\"></ds:Transform></ds:Transforms><ds:DigestMethod Algorithm=\"http://www.w3.org/2001/04/xmlenc#sha256\"></ds:DigestMethod><ds:DigestValue>em8csGAWynywpe8S4nN64o56/4DosXi2XWMY6RJ6YfA=</ds:DigestValue></ds:Reference></ds:SignedInfo>")
```

Comment:
Style: Using `%q{}` would get rid of all of the backslashes and not run the risk of accidentally introducing interpolation.